File: tests/nodejs/cases/case_016_stage_subscription.js
/**
 * Task case 016: stage_subscription
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case016StageSubscription(input) {
  // Task: stage subscription records and prepare transport-ready payload.
  const taskName = 'stage_subscription';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'subscription',
    stage: 'stage',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: payload is promoted to module-level retained cache for cross-request reuse.
  retainedCache.set('case_016', payload);
  return 'ok';
}

module.exports = {
  case016StageSubscription,
};

File: tests/go/cases/case_016_stage_subscription.go
package escape_tests

var retainedCase016 = []map[string]string{}

func Case016StageSubscription(input string) string {
    // Task case 016: stage subscription payload for downstream business processing.
    raw := input
    if raw == "" {
        raw = "sample"
    }
    payload := map[string]string{
        "task": "stage_subscription",
        "entity": "subscription",
        "stage": "stage",
        "input": raw,
    }
    // ESCAPE: payload is promoted to package-level retained cache.
    retainedCase016 = append(retainedCase016, payload)
    return "ok"
}
